src/nvim/eval.c

```diff
@@ -11635,6 +11632,7 @@ static void f_jobstart(typval_T *argvars, typval_T *rettv, FunPtr fptr)
 
   char **argv = tv_to_argv(&argvars[0], NULL);
   if (!argv) {
+    rettv->vval.v_number = -1;  // Return -1 on non executable
```

Comment:
jobstart() doc mentions:

```
Returns:
- 0 on invalid arguments or if the job table is full
- -1 if {cmd}[0] is not executable.
```

`tv_to_argv()` may fail for reasons other than non-executable, such as invalid arguments. So we need more granularity from `tv_to_argv()`.